src/routes/settings/components/Settings.js

```diff
@@ -0,0 +1,197 @@
+// @flow
+
+import * as React from 'react'
+import { SectionList, StyleSheet, Switch, AsyncStorage, Linking } from 'react-native'
+import styled from 'styled-components/native'
+
+import SettingItem from './SettingItem'
+import type { ThemeType } from '../../../modules/theme/constants/theme'
+import type { NavigationScreenProp } from 'react-navigation'
+import { reduce } from 'lodash/collection'
+import type { TFunction } from 'react-i18next'
+
+type PropsType = {
+  navigation: NavigationScreenProp<*>,
```

Comment:
Why is navigation needed?

Reply:
Settings will get the navigation prop because connect and withTheme passes it through. There are still some problems with the flow types but it works oke for now.